Wikipedia/Code/ProfileCoordinator.swift

```diff
@@ -104,8 +117,21 @@ class ProfileCoordinator: NSObject, Coordinator, ProfileCoordinatorDelegate {
     }
 
     func showDonate() {
-        // TODO
+        
+        guard let viewModel else {
+            return
+        }
+        
+        let donateCoordinator = DonateCoordinator(navigationController: navigationController, donateButtonGlobalRect: targetRects.donateButtonFrame, source: donateSouce, dataStore: dataStore, theme: theme, setLoadingBlock: { isLoading in
+            viewModel.isLoadingDonateConfigs = isLoading
+        })
+        
+        donateCoordinator.start()
+        
+        // Note: DonateCoordinator needs to handle a lot of delayed logic (fetch configs, present payment method action sheet, present native donate form and handle delegate callbacks from native donate form) as opposed to a fleeting navigation call with the other actions. For this reason we need to save it in a property so it isn't deallocated before this logic runs.
+        self.donateCoordinator = donateCoordinator
```

Comment:
If, in the future, we need to do this treatment to more coordinators, it could be worth doing "child coordinators." Here it works really well.